artemis-core-client/src/main/java/org/apache/activemq/artemis/core/client/impl/ClientMessageImpl.java

```diff
@@ -400,17 +398,10 @@ public long getLargeBodySize() {
       }
 
       @Override
-      public int encode(final ByteBuffer bufferRead) throws ActiveMQException {
-         ActiveMQBuffer buffer1 = ActiveMQBuffers.wrappedBuffer(bufferRead);
-         return encode(buffer1, bufferRead.capacity());
-      }
-
-      @Override
-      public int encode(final ActiveMQBuffer bufferOut, final int size) {
```

Comment:
I'm trying to understand why this was removed? I'm not saying it's wrong.. just trying to understand it. Can  you explain?

Reply:
Yes, it's a legit question :)
It has been removed because it isn't used anymore: the version (that already exists) with just ByteBuffers is enough for the purpose and it will save 3 wrapped instances (ChannelBufferWrapper+ 2 types of ByteBuf) that aren't needed by sequentialFile::writeDirect.
In addition it will make the code shorter  and with less checks about what ActiveMQBuffer contains (ie byte[])..